src/rustdoc-json-types/lib.rs

```diff
@@ -927,6 +927,16 @@ pub enum TraitBoundModifier {
     MaybeConst,
 }
 
+/// One precise capturing argument.
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum PreciseCapturingArg {
+    /// A lifetime.
+    Lifetime(String),
+    /// A non-lifetime argument such as a generic parameter or a constant parameter.
```

Comment:
Saying "such as" here can be read to mean there are other options not listed. Also, a "generic parameter" could refer to a constant parameter. It's clearer (to me at least) to say something like "A type or constant parameter".

I also don't love the name, but can't think of anything better off the top of my head. Curious if types people have any better ideas here. It does feel like we're leaking the HIR implementation details in that we differentiate lifetime, but not type vs generic parameters here.

Also: An example here would be nice.